src/scheduler/server_info.c

```diff
@@ -665,14 +664,19 @@ query_server_dyn_res(server_info *sinfo)
 
 			pipe_err = errno = 0;
 			/* Make sure file does not have open permissions */
-			err = tmp_file_sec_user(filename, 0, 1, S_IWGRP|S_IWOTH, 1, getuid());
-			if (err != 0) {
-				log_eventf(PBSEVENT_SECURITY, PBS_EVENTCLASS_SERVER, LOG_ERR, "server_dyn_res",
-					"error: %s file has a non-secure file access, setting resource %s to 0, errno: %d",
-					filename, res->name, err);
-				set_resource(res, res_zero, RF_AVAIL);
-				continue;
-			}
+			#if !defined(DEBUG) && !defined(NO_SECURITY_CHECK)
+				int err;
+				char *filename = conf.dynamic_res[i].script_name;
```

Comment:
You're defining variables not at the top of a block.  This will fail on earlier versions of C.  We generally stick with c89, which doesn't support this.  Put these back at the top and put a #if around them.

Reply:
really? we use things like \_\_func\__ and snprintf() which are C99 right?

Reply:
I believe we have had compilers in the past which were partially c99 compliant which allowed us to use that.  Until we're given the go-ahead to use c99, I'd shy away from it.  Unless @subhasisb disagrees.

Reply:
ok, that's fine, i was just hesitant to add an ugly #ifdef just to declare variables at the top of the block, I'll make the change.

Reply:
@agrawalravi90 you can do something like below (which is also ugly but will save u from another #ifdef):
```c
#if !defined(DEBUG) && !defined(NO_SECURITY_CHECK)
{
    int err;
    char *filename = conf.dynamic_res[i].script_name;
    ...
}
#endif
```